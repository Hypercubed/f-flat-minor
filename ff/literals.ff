
0b1111 . clr

0xFFFF . clr

0o27 . clr

1_000_000 . clr

0b1110_0000_0000_0000 . clr