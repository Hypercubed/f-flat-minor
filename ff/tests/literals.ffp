.import ../lib/core.ff
.import ../lib/testing.ffp

0 'Numeric' prints
    /* 0b1111 15 assert_eq
    0xFFFF 65535 assert_eq
    0o27 23 assert_eq
    1_000_000 1000000 assert_eq
    0b1110_0000_0000_0000 57344 assert_eq */

    /* -0b1111 -15 assert_eq
    -0xFFFF -65535 assert_eq
    -0o27 -23 assert_eq
    -1_000_000 -1000000 assert_eq
    -0b1110_0000_0000_0000 -57344 assert_eq */

    /* 1e10 10000000000 assert_eq
    -1e10 -10000000000 assert_eq */
OK

0 'Characters' prints 
    'A' 65 assert_eq
    'a' 97 assert_eq

    '\s' 32 assert_eq
    '\n' 10 assert_eq
OK

0 'Strings' prints 
  'abc' rot 'a' assert_eq 'b' assert_eq 'c' assert_eq
OK
