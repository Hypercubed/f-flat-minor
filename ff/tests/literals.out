Numeric OK
Characters OK
Strings OK
